src/canvg.js

```diff
@@ -1104,24 +1129,28 @@ function build(opts) {
       if (typeof ctx.font != 'undefined') {
         if (this.style('font').hasValue()) {
           ctx.font = this.style('font').value;
-          // store the font-size in case we have to update the current font-size
-          // we can add the element temporarily in dom to extract the style and parse the font easily
-          var element = doc.createElement('span');
-          element.setAttribute('style', 'font: ' + ctx.font);
-          this.styles['font-size'] = new svg.Property('font-size', element.style['fontSize']);
-          element.remove();
         } else {
+          var parentsFontSize;
+          if (this.parent) {
+            var parent = this.firstParentWithAbsoluteSize(this.parent, 'font-size');
```

Comment:
the absolute size is stored on `svg.emSize` for now ... look at how `getEM` works. all we have to do is tell the `toPixels` method whether we care about fontSize for %'s or not (can just be a bool). if we do, it should use the viewport, otherwise it should use the same logic as `getEM`. so i'd add a bool prop to `toPixels` and then set it to true in this general area but leave it false/undefined in other areas and that should be it.

Reply:
Thank you @gabelerner , I updated the PR according to your feedback. It's, indeed, much cleaner that way. Please let me know if you have more thoughts on it.